page_objects/shoping_cart_page.py

```diff
@@ -22,56 +22,44 @@ def __init__(self, driver):
     __complete_header_text = "THANK YOU FOR YOUR ORDER"
     __remove_from_shopping_cart_button = (By.XPATH, "//*[@id='remove-sauce-labs-backpack']")
 
-    def get_shopping_cart(self):
-        return self.__shopping_cart
+    def open_shopping_cart(self):
+        return self._click(self.__shopping_cart)
 
     def your_cart(self):
-        return self.__your_cart
+        return self.read_text(self.__your_cart)
 
     def your_cart_text(self):
         return self.__your_cart_text
 
-    def click_on(self, shopping_cart_item):
-        return self._click(shopping_cart_item)
-
-    def get_text(self, shopping_cart_item):
-        return self.read_text(shopping_cart_item)
-
-    def add_to_cart_button(self):
-        return self.__add_to_cart_button
+    def click_on_add_to_cart_button(self):
+        return self._click(self.__add_to_cart_button)
 
     def item_in_shopping_cart(self):
-        return self.__item_in_shopping_cart
+        return self.read_text(self.__item_in_shopping_cart)
 
     def name_of_item_in_shopping_item(self):
         return self.__name_of_item_in_shopping_item
 
-    def checkout_button(self):
-        return self.__checkout_button
+    def click_on_checkout_button(self):
+        return self._click(self.__checkout_button)
 
-    def set_first_name(self, first_name_value):
+    def set_credentials(self, first_name_value, last_name_value, postal_code_value):
         self._send_keys(self.__first_name_input, first_name_value)
-        return self
-
-    def set_last_name(self, last_name_value):
         self._send_keys(self.__last_name_input, last_name_value)
-        return self
-
-    def set_postal_code_input(self, postal_code_value):
         self._send_keys(self.__postal_code_input, postal_code_value)
         return self
 
-    def continue_button(self):
-        return self.__continue_button
+    def click_on_continue_button(self):
+        return self._click(self.__continue_button)
 
-    def finish_button(self):
-        return self.__finish_button
+    def click_on_finish_button(self):
+        return self._click(self.__finish_button)
 
     def complete_header(self):
-        return self.__complete_header
+        return self.read_text(self.__complete_header)
 
     def complete_header_text(self):
         return self.__complete_header_text
 
-    def remove_from_shopping_cart_button(self):
+    def click_on_remove_from_shopping_cart_button(self):
         return self.__remove_from_shopping_cart_button
```

Comment:
Why do you return the locator from the function click?  It makes no sence

Reply:
Fixed.